pkg/cli/sql_util.go

```diff
@@ -967,6 +936,79 @@ func runQueryAndFormatResults(conn *sqlConn, w io.Writer, fn queryFunc) (err err
 	}
 }
 
+// maybeShowTimes displays the execution time if show_times has been set.
+func maybeShowTimes(
+	conn *sqlConn, w io.Writer, isMultiStatementQuery bool, startTime,
+	queryCompleteTime time.Time,
+) {
+	if !sqlCtx.showTimes {
+		return
+	}
+
+	defer func() {
+		// If there was noticeable overhead, let the user know.
+		renderDelay := timeutil.Now().Sub(queryCompleteTime)
+		if renderDelay >= 1*time.Second && sqlCtx.isInteractive {
+			fmt.Fprintf(stderr,
+				"\nNote: an additional delay of %s was spent formatting the results.\n"+
+					"You can use \\set display_format to change the formatting.\n",
+				renderDelay)
+		}
+		// An additional empty line as separator.
+		fmt.Fprintln(w)
+	}()
+
+	clientSideQueryTime := queryCompleteTime.Sub(startTime)
+	// We don't print timings for multi-statement queries as we don't have an
+	// accurate way to measure them currently. See #48180.
+	if isMultiStatementQuery {
+		// No need to print if no one's watching.
+		if sqlCtx.isInteractive {
+			fmt.Fprintf(stderr, "\nNote: timings for multiple statements on a single line are not supported. See %s.\n",
+				unimplemented.MakeURL(48180))
+		}
+		return
+	}
+
+	// Print a newline early. This provides a discreet visual
+	// feedback that execution finished, and that the next line of
+	// output will be execution time(s).
+	fmt.Fprintln(w)
+
+	if sqlCtx.verboseTimings {
+		fmt.Fprintf(w, "Time: %s", clientSideQueryTime)
+	} else {
+		// Simplified displays: human users typically can't
+		// distinguish sub-millisecond latencies.
+		fmt.Fprintf(w, "Time: %.3fs", clientSideQueryTime.Seconds())
+	}
+
+	if !sqlCtx.enableServerExecutionTimings {
+		fmt.Fprintln(w)
+		return
+	}
+
+	// If discrete server/network timings are available, also print them.
+	execLatency, serviceLatency, err := conn.getLastQueryStatistics()
+	if err != nil {
+		fmt.Fprintf(stderr, "\nwarning: %v", err)
+		return
+	}
+
+	networkLatency := clientSideQueryTime - serviceLatency
+	if sqlCtx.verboseTimings {
+		fmt.Fprintf(w, " total (exec %s / net %s / other %s)\n",
```

Comment:
it could be nice to document what we mean by "exec" "net" and "other" but i don't know where we would convey this information to the user. definitely not a reason to block this backport anyway.

Reply:
yes I agree with this!

but that seems to be more a point for documentation